src/csharp/Microsoft.Spark/Utils/VersionSensor.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Types;
+
+namespace Microsoft.Spark.Utils
+{
+    /// <summary>
+    /// Gets the <see cref="VersionInfo"/> for the "Microsoft.Spark" and "Microsoft.Spark.Worker"
+    /// assemblies if they exist within the current execution context of this application domain.
+    /// </summary>
+    internal class VersionSensor
+    {
+        private const string MicrosoftSparkAssemblyName = "Microsoft.Spark";
+        private const string MicrosoftSparkWorkerAssemblyName = "Microsoft.Spark.Worker";
+
+        private static readonly Lazy<VersionInfo> s_microsoftSparkVersionInfo =
+            new Lazy<VersionInfo>(() => CreateVersionInfo(MicrosoftSparkAssemblyName));
+
+        private static readonly Lazy<VersionInfo> s_microsoftSparkWorkerVersionInfo =
+            new Lazy<VersionInfo>(() => CreateVersionInfo(MicrosoftSparkWorkerAssemblyName));
+
+        internal static VersionInfo MicrosoftSparkVersion() => s_microsoftSparkVersionInfo.Value;
+
+        internal static VersionInfo MicrosoftSparkWorkerVersion() =>
+            s_microsoftSparkWorkerVersionInfo.Value;
+
+        private static VersionInfo CreateVersionInfo(string simpleAssemblyName)
```

Comment:
just `assemblyName`? why simple?

Reply:
`AssemblyName` has the `Name` property which is the simple name of the assembly and the `FullName` property which is the full name of the assembly.  But I'll change this var name to what you've suggested.